Add Forwarder upgrade troubleshooting details for Step Functions Tracing (#21681)
DataDog/documentation: content/en/serverless/step_functions/troubleshooting.md

- Check the Datadog Lambda Forwarder for error messages. Ensure that you have correctly set your API key and Datadog site.
- Enable `DEBUG` logs on the Datadog Lambda Forwarder by setting the environment variable `DD_LOG_LEVEL` to `debug`.

### Verify that logs are searchable on Live Search and have DD_TRACE_ENABLED tag

> In Datadog, go to [**Logs > Live Tail**][2]. Search for `source:stepfunction`. You may need to trigger the state machine a few times. If you need to upgrade Datadog Lambda Forwarder from an older version, check that after the upgrade the Forwarder has `DD_FETCH_STEP_FUNCTIONS_TAGS` tag and the value is set to `true`. If the upgraded Forwarder does not have `DD_FETCH_STEP_FUNCTIONS_TAGS` tag, your Forwarder may not be upgraded correctly. 

Reviewer: ```suggestion
In Datadog, go to [**Logs > Live Tail**][2]. Search for `source:stepfunction`. You may need to trigger the state machine a few times. If you need to upgrade Datadog Lambda Forwarder from an older version, check that after the upgrade, the Forwarder has the `DD_FETCH_STEP_FUNCTIONS_TAGS` tag set to `true`. If the upgraded Forwarder does not have the `DD_FETCH_STEP_FUNCTIONS_TAGS` tag, your Forwarder may not be upgraded correctly. 
```

> If the Forwarder and state machine tag is set up correctly with the aformentioned steps, the logs should be tagged with `DD_TRACE_EANBLED:true` tag.

Reviewer: ```suggestion
If the Forwarder and state machine tags are set up correctly with the previous steps, the logs are tagged with `DD_TRACE_EANBLED:true`.
```
